File: php/classes/padding/PKCS7.php
<?php

class PKCS7 
{
	/**
	* Pads the text with PKCS#7 padding scheme.
	*/
	public static function pad($text, $size)
	{
		$pad = $size - (strlen($text) % $size);
		return $text . str_repeat(chr($pad), $pad);
	}
	
	/**
	* Unpads the text from PKCS#7 padding scheme.
	*/
	public static function unpad($text)
	{
		$pad = ord($text{strlen($text) - 1});
		if ($pad > strlen($text)) return false;
		if (strspn($text, chr($pad), strlen($text) - $pad) != $pad) return false;
		return substr($text, 0, -1 * $pad);
	}

}

?>

File: csharp/ASCrypt/Padding/PKCS7.cs
using System;

namespace ASCrypt.Padding
{
    public class PKCS7
    {
        /// <summary>
        /// Private error message constants of the class.
        /// </summary>
        private static readonly String ERROR_VALUE = "Invalid padding value. Got {0}, expected {1}.";
        
        /// <summary>
        /// Pads the bytes with PKCS#7 padding scheme.
        /// </summary>
        public static Byte[] Pad(Byte[] bytes, Int32 size)
        {
            Byte[] c = (Byte[])bytes.Clone();
            Int32 s = size - c.Length % size;
            for (Int32 i = 0; i < s; i++)
            {
                Array.Resize(ref c, c.Length + 1);
                c[c.Length - 1] = (Byte)s;
            }
            return c;
        }

        /// <summary>
        /// Unpads the bytes with PKCS#7 padding scheme.
        /// </summary>
        public static Byte[] Unpad(Byte[] bytes)
        {
            Byte[] c = (Byte[])bytes.Clone();
            Byte s = (Byte)c[c.Length - 1];
            for (Int32 i = s; i > 0; i--)
			{
				Int32 v = c[c.Length - 1];
                Array.Resize(ref c, c.Length - 1);
				if (s != v) 
				{
                    String msg = String.Format(ERROR_VALUE, v, s);
                    throw new Exception(msg);
				}
			}
            return c;
        }

    }

}
